packages/react/src/hooks/useUserAttributes.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ *  This version of useUserAttributes is functional, but there are several type errors relating to the handleActions. Also, the handleActions are hardcoded.
+ */
+
+import { Hub, HubCallback } from "@aws-amplify/core";
+
+import { useDataState } from "./useDataState";
+import {
+  DefaultAttributes,
+  defaultSendUserAttributeVerificationCodeOutput,
+  defaultUpdateUserAttributesOutput,
+} from "./constants";
+
+import {
+  HandlerInputs,
+  StateDataTypes,
+  confirmUserAttributeAction,
+  deleteUserAttributesAction,
+  fetchUserAttributesAction,
+  sendUserAttributeVerificationCodeAction,
+  updateUserAttributesAction,
+  Actions,
+} from "./actionTypes";
+import React from "react";
+
+interface ActionState<T> {
+  /**
+   * action data
+   */
+  data: T;
+  /**
+   * indicates whether action is running
+   */
+  isLoading: boolean;
+  /**
+   * error message
+   */
+  message: string | undefined;
+}
+
+// interface Actions {
+//   confirm: typeof confirmUserAttributeAction;
+//   delete: typeof deleteUserAttributesAction;
+//   sendVerificationCode: typeof sendUserAttributeVerificationCodeAction;
+//   update: typeof updateUserAttributesAction;
+//   fetch: typeof fetchUserAttributesAction;
+// }
+
+const useUserAttributes = <T extends keyof Actions>(
+  action: T
+): [
+  state: ActionState<StateDataTypes[T]>,
+  handleAction: (...input: HandlerInputs[T][]) => void
+] => {
+  const [deleteState, handleDelete] = useDataState(
+    deleteUserAttributesAction,
+    undefined
+  );
```

Comment:
Let's break these in to individual hooks to isolate unit testing and recompose them here. Example:
```ts
 const useDeleteUserAttributes = useDataState(
    deleteUserAttributesAction,
    undefined
  );

const useUserAttributes = <T extends keyof Actions>(
  action: T
): [
  state: ActionState<StateDataTypes[T]>,
  handleAction: (...input: HandlerInputs[T][]) => void
] => {
  const delete = useDeleteUserAttributes();
  const update = useUpdateUserAttributes();
  // ...etc
```